donut/modules/newsgroups/helpers.py

```diff
@@ -209,6 +215,7 @@ def get_owners(group_id):
 def get_posting_positions(group_id, user_id):
     """Get positions user can send as in a group."""
 
+    pos = auth_utils.get_admin_pos(user_id)
```

Comment:
Hmm, to reduce code duplication maybe it makes sense for all these queries to return `pos_id, group_name, pos_name` and then to concatenate the group and position names in Python.